Calculator.Wpf/Validation/UserValidator.cs

```diff
@@ -0,0 +1,186 @@
+                CheckOpeningBracketAfterNumber(expression)
+            };
+
+            foreach (string errorMessage in errorMessages)
+            {
+                if (!string.IsNullOrEmpty(errorMessage))
+                {
+                    return errorMessage;
+                }
+            }
+
+            return "Expression is valid";
+        }
+
+        private string CheckDivisionByZero(string expression)
+        {
+            if (expression.Contains("/0"))
+            {
+                return "Division by zero detected";
+            }
+            return null;
+        }
+
+        private string CheckInvalidVariableExpression(string expression)
+        {
+
+            return ProcessExpression(expression, (currentChar, nextChar) =>
+            {
+                if (char.IsDigit(currentChar) && char.IsLetter(nextChar))
+                {
+                    return $"Invalid variable expression '{currentChar}{nextChar}' found";
+                }
+                else if (char.IsLetter(currentChar) && char.IsDigit(nextChar))
+                {
+                    return $"Invalid variable expression '{currentChar}{nextChar}' found";
+                }
+
+                return null;
+            });
+        }
+
+        private string CheckInvalidVariableSequence(string expression)
+        {
+            return ProcessExpression(expression, (currentChar, nextChar) =>
+            {
+                if (char.IsLetter(currentChar) && (nextChar == '.' || nextChar == ',' || nextChar == '(' || nextChar == ')'))
+                {
+                    return $"Invalid variable expression '{currentChar}{nextChar}' found";
+                }
+                else if ((currentChar == '.' || currentChar == ',') && char.IsLetter(nextChar))
+                {
+                    return $"Invalid variable expression '{currentChar}{nextChar}' found";
+                }
+
+                return null;
+            });
+        }
+
+        private string CheckUnmatchedParentheses(string expression)
+        {
+            int openParenthesesCount = expression.Count(c => c == '(');
+            int closingParenthesesCount = expression.Count(c => c == ')');
+
+            if (openParenthesesCount != closingParenthesesCount)
+            {
+                if (openParenthesesCount > closingParenthesesCount)
+                {
+                    return "Unmatched opening parenthesis";
+                }
+                else
+                {
+                    return "Unmatched closing parenthesis";
+                }
+            }
+
+            return null;
+        }
+
+        private string CheckCorrectSequenceОfStatementsOperators(string expression)
+        {
+            return ProcessExpression(expression, (currentChar, nextChar) =>
+            {
+                if ((currentChar == '+' || currentChar == '-' || currentChar == '*' || currentChar == '/' || currentChar == '(' || currentChar == ')')
```

Comment:
it is be more readable if you will create array with all elements like '+', '-' and so on. and in if will use Contains to check if there is currentChar in array. check this in other methods